### PR TITLE
Avoid encoding between utf-8 and 16 unnecessarily

### PR DESCRIPTION
**Pull request by bratseth**
@arnej27959 it's back!

Once this is verified I'll do the same for the XML renderer. Ideally we should do all of this for struct fields as well.

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/prelude/fastsearch/DocsumField.java [hunk 1]
```diff
@@ -96,4 +96,7 @@ public String getName() {
      */
     public abstract Object convert(Inspector value);
 
+    /** Returns whether this is the string field type. */
+    boolean isString() { return false; }
```
Comment (arnej27959) [inline container-search/src/main/java/com/yahoo/prelude/fastsearch/DocsumField.java]: could also be "hasUtf8()" to reflect intended usage

File: container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java [hunk 2]
```diff
@@ -787,5 +666,156 @@ void setGenerator(JsonGenerator generator) {
...
+
+        private MutableBoolean hasFieldsField;
+
+        public FieldConsumer(JsonGenerator generator, boolean debugRendering) {
+            this.generator = generator;
+            this.debugRendering = debugRendering;
+        }
+
+        /**
+         * Called before each use of this for a hit to keep track of whether we
```
Comment (arnej27959) [inline container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java]: the "setCurrent" is called at the start of rendering each hit, right?
and then "fieldsStart" is called for every field?

**Review by arnej27959 — APPROVED** [review]
looks good to me!

File: container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java (hunk 2 above)
Comment (bratseth) [reply container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java]: Yes.

**Pull request merged**